Src/ILGPU/Runtime/ArrayViews.tt

```diff
@@ -268,10 +270,11 @@ namespace ILGPU.Runtime
         {
 <#      for (int i = 0; i < dim.Dimension; ++i) { #>
 <#          var iName = IndexDimensions[i].PropertyName; #>
+<#          var thisExtent = $"({baseType})Extent.{iName}"; #>
             Trace.Assert(
-                index.<#= iName #> >= 0 &
-                index.<#= iName #> + extent.<#= iName #> <=
-                    (<#= baseType #>)Extent.<#= iName #>,
+                index.<#= iName #> >= 0 &&
```

Comment:
nit: can we change this to a logic `|` operation in order to improve compilation speed in debug builds?